emloop/api.py

```diff
@@ -172,51 +155,51 @@ def create_hooks(config: dict, model: Optional[AbstractModel]=None, dataset: Opt
     :param output_dir: training output dir available to the hooks
     :return: list of hook objects
     """
-    try:
-        logging.info('Creating hooks')
-        config = copy.deepcopy(config)
-        hooks = []
-        if 'hooks' in config:
-            for hook_config in config['hooks']:
-                if isinstance(hook_config, str):
-                    hook_config = {hook_config: {}}
-                assert len(hook_config) == 1, 'Hook configuration must have exactly one key (fully qualified name).'
-
-                hook_path, hook_params = next(iter(hook_config.items()))
-                if hook_params is None:
-                    logging.warning('\t\t Empty config of `%s` hook', hook_path)
-                    hook_params = {}
-
-                hook_module, hook_class = parse_fully_qualified_name(hook_path)
-                # find the hook module if not specified
-                if hook_module is None:
-                    hook_module = get_class_module(EL_HOOKS_MODULE, hook_class)
-                    logging.debug('\tFound hook module `%s` for class `%s`', hook_module, hook_class)
-                    if hook_module is None:
-                        raise ModuleNotFoundError('Can`t find hook module for hook class `{}`. '
-                                                  'Make sure it is defined under `{}` sub-modules.'
-                                                  .format(hook_class, EL_HOOKS_MODULE))
-                # create hook kwargs
-                hook_kwargs = {'dataset': dataset, 'model': model, 'output_dir': output_dir, **hook_params}
-
-                # create new hook
-                try:
-                    hook = create_object(hook_module, hook_class, kwargs=hook_kwargs)
-                    hooks.append(hook)
-                    logging.info('\t%s created', type(hooks[-1]).__name__)
-                except (ValueError, KeyError, TypeError, NameError, AttributeError, AssertionError, ImportError) as ex:
-                    logging.error('\tFailed to create a hook from config `%s`', hook_config)
-                    raise ex
-    except Exception as ex:
-        logging.error('Creating hooks failed')
-        raise ex
+    logging.info('Creating hooks')
+    hooks_config = copy.deepcopy(config.get("hooks", {}))
+    hooks = []
+
+    if "TrainingTrace" not in hooks_config:
```

Comment:
This is not a reliable way to check it, it can be `'emloop.hooks.TrainingTrace` or pretty much anything (where it is imported). We need to find a better way.